admin/class-admin-init.php

```diff
@@ -130,6 +131,86 @@ private function get_yoast_seo_suggested_plugins_notification( $name, $plugin, $
 		);
 	}
 
+	/**
+	 * Determines whether a update notification needs to be displayed.
+	 *
+	 * @return void
+	 */
+	public function yoast_plugin_update_notification() {
+		$notification_center = Yoast_Notification_Center::get();
+		$current_version     = WPSEO_Options::get( 'version', WPSEO_VERSION );
+
+		$file = plugin_dir_path( WPSEO_FILE ) . 'release-info.json';
+		$release_json = file_get_contents( $file );
+		/**
+		 * Filter: 'wpseo_update_notice_content' - Allow filtering of the content
+		 * of the update notice read from the release-info.json file.
+		 *
+		 * @api object The object from the release-info.json file.
+		 */
+		$release_info = apply_filters( 'wpseo_update_notice_content', json_decode( $release_json ) );
+
+		// Remove if file is not present, malformed or for a different version.
+		if ( is_null( $release_info )
+			|| empty( $release_info->version )
+			|| (float) $release_info->version !== (float) $current_version
+		 	|| empty( $release_info->release_description )
+		) {
+			$notification_center->remove_notification_by_id( 'wpseo-plugin-updated' );
+			return;
+		}
+
+		$notification  = $this->get_yoast_seo_update_notification( $release_info );
+		$dismissal_key = $notification->get_dismissal_key();
+
+		// Restore notification if it was dismissed in a previous minor version.
+		$last_dismissed_version = get_user_option( $dismissal_key );
+		if ( ! $last_dismissed_version || (float) $last_dismissed_version < (float) $current_version ) {
+			Yoast_Notification_Center::restore_notification( $notification );
+		}
+		$notification_center->add_notification( $notification );
+	}
+
+	/**
+	 * Build Yoast SEO update notification.
+	 *
+	 * @param object $release_info The release information.
+	 *
+	 * @return Yoast_Notification The notification for the present version
+	 */
+	private function get_yoast_seo_update_notification( $release_info ) {
+		$info_message = '<strong>' .
+				sprintf(
+				/* translators: %1$s expands to Yoast SEO, %2$s expands to the plugin version. */
+					__( 'New in %1$s %2$s: ', 'wordpress-seo' ),
+					'Yoast SEO',
+					$release_info->version
+				) .
+				'</strong>' .
+				$release_info->release_description;
+		if ( ! empty( $release_info->shortlink ) ) {
+			$link = esc_url( WPSEO_Shortlinker::get( $release_info->shortlink ) );
+			$info_message .= ' <a href="' . esc_url( $link ) . '" target="_blank">' .
+							 sprintf(
+							 	/* translators: %s expands to the plugin version. */
+							 	__( 'Read all about version %s here', 'wordpress-seo' ),
+								$release_info->version
+							 ) .
+							 '</a>';
+		}
+		$data         = (object) [ 'dismiss_value' => WPSEO_Options::get( 'version', WPSEO_VERSION ) ];
```

Comment:
There doesn't seem to be a need to cast this array to an object first for it to be properly encoded later on by `wp_json_encode`. Additionally, you could directly pass it along on line 208 instead of assigning it to a variable.